### PR TITLE
Dogstatsd unit test stabilization for mac

### PR DESCRIPTION
**Pull request by ddrthall**
### What does this PR do?
Restructures tests for dogstatsd server and listeners to minimize dependencies on uds/udp connections in unit tests. This will hopefully help stabilize the tests on the mac gitlab runners.

### Possible Drawbacks / Trade-offs
The following test functions were removed:

func TestScanLines(t *testing.T) & func TestEOLParsing(t *testing.T)
These tests were merged in to TestNextMessage, since they were focused on functions that were judged to be a bit too close to "implementation detail"

func TestStaticTags(t *testing.T)
TestNewServerExtraTags tests to confirm the pickup of the "tags" field in fargate and its inclusion in extra tags, and TestExtraTags tests to confirm that extra tags is utilized when it should be

func TestE2EParsing(t *testing.T)
Integration tests cover "e2e", and newline separators in the same message are accounted for in TestMetricPermutations

### CODE REVIEW

**Review by gh123man — COMMENTED** [review]
Just some minor comments/questions. These changes look great.

File: comp/dogstatsd/listeners/uds_common_test.go [hunk 1]
```diff
@@ -122,3 +126,102 @@ func testStartStopUDSListener(t *testing.T, listenerFactory udsListenerFactory,
...
+		return conn.readStream(b)
+	}
+	return conn.readDatagram(b)
+}
+
+func (conn *mockUnixConn) readDatagram(b []byte) (int, error) {
+	if conn.offset >= len(conn.buffer) {
+		select {
+		case <-conn.stop:
+		case <-time.After(time.Second * 2):
```
[the remark this answers is not in the document]
Comment (ddrthall) [reply comp/dogstatsd/listeners/uds_common_test.go]: I'm going with an explicit error return - we'll see that logged by the caller in a true error case. Failing tests in threads seems like it's pretty wonky, especially if we're just running a single test in isolation and expect the process to exit quick.

**Review by gh123man — APPROVED** [review]
Nice work 👌 
Sorry it took so long to get this reviewed!

**Pull request merged**